src/providers/hover.ts

```diff
@@ -1,47 +1,60 @@
 import * as vscode from 'vscode';
 import * as state from '../state';
-import { nClient as nClient } from "../connector";
 import * as util from '../utilities';
 
 export default class HoverProvider implements vscode.HoverProvider {
     state: any;
+
     constructor() {
         this.state = state;
     }
-    formatDocString(nsname, arglist, doc) {
+
+    formatDocString(nsname: string, arglist: string, documentation: string) {
+
         let result = '';
+        // Format the namespace.
         if (nsname.length > 0 && nsname !== 'undefined') {
             result += '**' + nsname + '**  ';
             result += '\n';
         }
-
         // Format the different signatures for the fn
-        if (arglist & arglist.substring) {
+        if (arglist && arglist != "") {
             result += arglist.substring(0, arglist.length)
-                .replace(/\]/g, ']_')
-                .replace(/\[/g, '* _[');
+                .replace(/\]/g, ']')
+                .replace(/\[/g, '* [');
+            result += '\n\n';
             result += '\n\n';
         }
-
         // Format the actual docstring
-        if (doc) {
-            result += doc.replace(/\s\s+/g, ' ');
+        if (documentation && documentation != "") {
```

Comment:
Jumping in here, so please disregard.
This is basically the same as https://github.com/BetterThanTomorrow/calva/pull/375/files#diff-9205a2ccff45e74e74a392042d83efe8R25 Could/should it be extracted into a `docUtils` (or some such, naming is hard)?